tests/testthat/test-densities.R

```diff
@@ -52,64 +52,83 @@ test_that("dbinom agrees", {
 
 
 test_that("dnbinom agrees", {
-  size <- as.numeric(1:50)
-  prob <- runif(length(size))
-  mu <- size * (1 - prob) / prob
-  x <- as.integer(sample(size, replace = TRUE))
-  expect_equal(dust_dnbinom(x, size, mu, TRUE),
-               dnbinom(x, size, mu = mu, log = TRUE))
-  expect_equal(dust_dnbinom(x, size, mu, FALSE),
-               dnbinom(x, size, mu = mu, log = FALSE))
-
-  ## size > x case which was implemented incorrectly in <= v0.6.5
-  expect_equal(
-    dnbinom(511, 2, mu = 6.65, log = TRUE),
-    dust_dnbinom(511L, 2, 6.65, TRUE))
-
-  ## Allow non integer size (wrong in <= 0.7.5)
-  expect_equal(
-    dust_dnbinom(x = 511L, size = 3.5, mu = 1, log = TRUE),
-    dnbinom(511, 3.5, mu = 1, log = TRUE))
+  for (is_float in c(FALSE, TRUE)) {
+    if (is_float) {
+      tolerance <- sqrt(sqrt(.Machine$double.eps))
```

Comment:
Why sqrt twice?

Reply:
I might be getting this wrong but I find that:
sqrt(flt_epsilon) = sqrt(2^-23) = 0.0003449638
sqrt(sqrt(dbl_epsilon) = sqrt(sqrt(2^-52)) = 0.0001220703

Reply:
flt_eps is ~1e-8 (https://en.wikipedia.org/wiki/Machine_epsilon), double is ~1e-16

Reply:
Seems fine to round those, but I don't understand the rationale for writing the approximation as flt_eps =~ sqrt(dbl_eps) which I understand as being roughly true by chance, rather than just writing flt_eps as a constant. Is it expected to be machine/implementation specific?

Reply:
(this doesn't matter for the code though, just asking out of interest)

Reply:
just avoiding magic numbers and writing everything in terms of eps :)